homeassistant/components/ledsc/light.py

```diff
@@ -0,0 +1,175 @@
+"""LedSC light."""
+
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.components.light import ColorMode, LightEntity
+from homeassistant.core import HomeAssistant
+
+from websc_client import WebSClientAsync as WebSClient
+from websc_client import WebSCAsync
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(
+    hass: HomeAssistant, config, add_entities: AddEntitiesCallback, discovery_info=None
+):
+    """Redirects to '__setup'."""
+    hass.async_create_task(__setup(hass, dict(config), add_entities))
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config: ConfigEntry, add_entities: AddEntitiesCallback
+):
+    """Redirects to '__setup'."""
+    await __setup(hass, dict(config.data), add_entities)
+
+
+async def __setup(hass: HomeAssistant, config: dict, add_entities: AddEntitiesCallback):
+    """
+    Connect to WebSC.
+
+    load the configured devices and add them to hass.
+    """
+    host = config["host"]
+    port = config["port"]
+
+    client = WebSClient(host=host, port=port)
+    await client.connect()
+    hass.async_create_background_task(client.observer(), name="ledsc-observer")
+
+    devices: list[LedSC] = list()
+    for websc in client.devices.values():
+        ledsc = LedSC(
+            client_id=f"{host}:{port}",
+            websc=websc,
+            hass=hass,
+        )
+        websc.set_callback(__generate_callback(ledsc))
+        devices.append(ledsc)
+    add_entities(devices, True)
+
+
+class LedSC(LightEntity):
+    """Representation of an Awesome Light."""
+
+    def __init__(
+        self,
+        client_id: str,
+        websc: WebSCAsync,
+        hass: HomeAssistant,
+    ) -> None:
+        """Initialize an AwesomeLight."""
+        self._hass: HomeAssistant = hass
+        self._websc: WebSCAsync = websc
+        self.__id = f"{client_id}-{websc.name}"
+        _LOGGER.info(f"LedSC '%s' initialized!", self.name)
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return id unique for client and entity name combination."""
+        return self.__id
+
+    @property
+    def supported_color_modes(self) -> set[ColorMode] | set[str] | None:
+        """List of supported color modes."""
+        return {ColorMode.RGBW}
+
+    @property
+    def color_mode(self) -> ColorMode | str | None:
+        """Return the current color mode (static)."""
+        return ColorMode.RGBW
+
+    @property
+    def available(self) -> bool:
+        """
+        Check if light is available.
+
+        The information is from WebSC.
+        """
+        return not self._websc.is_lost
+
+    @property
+    def name(self) -> str:
```

Comment:
Use the entity_description attribute instead and use strings.json for device names instead

Reply:
I can use `entity_description` for more detailed device information, but the entity name is dynamic based on the WebSC configuration.

Reply:
I think you don't even need a name for the entity, if you leave it empty, and set the name of the device and the manufacturer in the DeviceInfo it will use the generic name "Light", the entity will then be named light.ledsc_devicename_light

Reply:
![image](https://github.com/home-assistant/core/assets/57815256/44950f2d-3ff5-4f70-abcc-3884c9a1db5b)

Do is not a very nice default name.
If you don't mind, I would like to leave this property. This name is set in WebSC.
![Snímek obrazovky z 2024-05-30 18-33-19](https://github.com/home-assistant/core/assets/57815256/4358baf6-59bb-4832-82c6-b9f0731cd2bc)

Reply:
It's tricky to get this right, haven't done this yet with the light platform. I would like to test it, is there some way to do this? The host demo.ledsc.eu did not work.

Reply:
The demo.ledsc.eu domain is still under development. I am currently creating a way to create a virtual instance of WebSC so that anyone who wants to try the demo can run their own instance of WebSC for a limited time. However, for testing purposes I can run one static instance on this domain. Otherwise you can run your own instances of course.

Reply:
You can try: http://demo.ledsc.eu:8080/

Reply:
nope, server rejects the connection